contrib/icinga2-commands.conf

```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
             description = "Look for ec2 instance by id"
         }
         "-n" = {
-            value = "$cloud_aws_name$"
+            value = "$cloud_aws_ec2_instance_name$"
             description = "Look for ec2 instance by name"
         }
     }
-
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instance = "instance"
 }
 
-object CheckCommand "cloud_aws_instance" {
+object CheckCommand "cloud_aws_instances" {
     import "cloud_aws"
 
+    command += [ "ec2", "instances" ]
+
     arguments += {
-        "ec2" = {
-            value = "$cloud_aws_ec2$"
-            description = "Checks in the EC2 context"
-        }
-        "instances" = {
-            value = "$cloud_aws_instances$"
-            description = "Checks multiple EC2 instances by name"
-        }
         "-n" = {
-            value = "$cloud_aws_name$"
+            value = "$cloud_aws_ec2_instances_name$"
             description = "Search for ec2 instances by name (e.g. instance*)"
         }
         "-a" = {
-            value = "$cloud_aws_autoscale$"
+            value = "$cloud_aws_ec2_instances_autoscale$"
             description = "Search for ec2 instances by autoscaling group"
         }
     }
+}
+
+object CheckCommand "cloud_aws_cloudfront" {
+    import "cloud_aws"
+
+    command += [ "cloudfront" ]
+
+    arguments += {
+        "-e" = {
+            value = "$cloud_aws_etag$"
+            description = "Etag name of one or multiple distributions. If --etag is empty, all distributions will be evaluated"
+        }
+    }
+}
+
+object CheckCommand "cloud_aws_s3_bucket" {
+    import "cloud_aws"
+
+    command += [ "s3", "bucket" ]
 
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instances = "instances"
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_bucket_buckets$"
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
+        }
+        "-c" = {
+            value = "$cloud_aws_s3_bucket_crit_bucket_size$"
+            description = "Critical threshold for the size of the specified bucket. Alerts if the size is greater than the critical threshold"
+        }
+        "-w" = {
+             value = "$cloud_aws_s3_bucket_warn_bucket_size$"
+             description = "Warning threshold for the size of the specified bucket. Alerts if the size is greater than the warning threshold"
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_bucket_empty_ok$"
+            description = "Return OK if there are no buckets"
+        }
+    }
 }
 
-apply Service "cloud_aws_single_instance" {
+object CheckCommand "cloud_aws_s3_object" {
+    import "cloud_aws"
+
+    command += [ "s3", "object" ]
+
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_object_bucket$"
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
```

Comment:
`--buckets` makes no sense in the description as the user should not see this, especially when -b is used as argument. Better use the value `cloud_aws_s3_object_bucket` or something like "If no bucket is given, ...".